### PR TITLE
Add RNA transcription exercise

### PR DESCRIPTION
**Pull request by clementi**

### CODE REVIEW

File: exercises/rna-transcription/test/test_rna_transcription.c [hunk 1]
```diff
@@ -0,0 +1,45 @@
+#include "vendor/unity.h"
+#include "../src/rna_transcription.h"
+#include <stdlib.h>
+#include <string.h>
+
+void resetTest(void) {}
```
Comment (ryanplusplus) [inline exercises/rna-transcription/test/test_rna_transcription.c]: This looks to be unused. Can you remove it?

File: exercises/rna-transcription/src/example.c [hunk 2]
```diff
@@ -0,0 +1,27 @@
+#include <stdlib.h>
+#include <string.h>
+#include "rna_transcription.h"
+
+char *to_rna(const char *dna) {
+  int len = strlen(dna);
```
Comment (ryanplusplus) [inline exercises/rna-transcription/src/example.c]: `strlen

File: exercises/rna-transcription/src/example.c [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  char *rna = (char *)malloc(sizeof(char) * len);
```
Comment (ryanplusplus) [inline exercises/rna-transcription/src/example.c]: Can you remove this unnecessary cast? See http://stackoverflow.com/questions/605845/do-i-cast-the-result-of-malloc.

File: exercises/rna-transcription/test/test_rna_transcription.c [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+void test_transcribes_A_to_U(void) {
+  test_transcription("A", "U");
+}
+
+void test_transcribes_all_occurrences(void) {
+  test_transcription("ACGTGGTCTTAA", "UGCACCAGAAUU");
+}
+
```
Comment (ryanplusplus) [inline exercises/rna-transcription/test/test_rna_transcription.c]: There are some tests missing from the [common test suite for this exercise](https://github.com/exercism/x-common/blob/master/exercises/rna-transcription/canonical-data.json). Can you add the tests related to input validation?

**Comment by ryanplusplus** [comment]
Just a couple of nitpicks from me. Thanks for contributing!

**Comment by clementi** [comment]
@ryanplusplus Sure. Fixes committed.

**Comment by ryanplusplus** [comment]
@clementi looks good to me except the the wrong type is still used for `strlen`. I'll merge once that's addressed. Thanks again!

**Comment by clementi** [comment]
Fix committed.

**Pull request merged**